tests/internals/initial_delay_cooldownperiod/initial_delay_cooldownperiod_test.go

```diff
@@ -0,0 +1,128 @@
+//go:build e2e
+// +build e2e
+
+package initial_delay_cooldownperiod_test
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+const (
+	testName = "initial-delay-cooldownperiod-test"
+)
+
+var (
+	testNamespace    = fmt.Sprintf("%s-ns", testName)
+	deploymentName   = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName = fmt.Sprintf("%s-so", testName)
+
+	now   = time.Now().Local()
+	start = (now.Minute() + 10) % 60
+	end   = (start + 1) % 60
+)
+
+type templateData struct {
+	TestNamespace    string
+	DeploymentName   string
+	ScaledObjectName string
+	StartMin         string
+	EndMin           string
+}
+
+const (
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    deploy: {{.DeploymentName}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      pod: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        pod: {{.DeploymentName}}
+    spec:
+      containers:
+        - name: nginx
+          image: 'nginxinc/nginx-unprivileged'
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  cooldownPeriod: 5
+  minReplicaCount: 0
+  initialDelayCooldownPeriod: 120
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 15
+  triggers:
+  - type: cron
+    metadata:
+      timezone: Etc/UTC
+      start: {{.StartMin}} * * * *
+      end: {{.EndMin}} * * * *
+      desiredReplicas: '10'
+`
```

Comment:
Do we need an active scaler? I mean, just a cron scaler with any cron conditions and `desiredReplicas: '0'` can work in this case as we want to test initial cooldown, so it seems a better choice to ensure that we force the new condition.

Reply:
Sorry, I don't  understand what you mean.

Reply:
Sorry, I don't understand what you mean @JorTurFer